### PR TITLE
Added new options for network, zone, machine type, firewall rule creation and rm

### PR DESCRIPTION
**Pull request by coleca**
Also fixed issue with gce parms

### CODE REVIEW

**Comment by cloudnautique** [comment]
Thanks! I'll test this weekend, the changes will make it more usable by non-rancher folks :)

**Comment by coleca** [comment]
Fixed typo on firewall rule create

File: bin/gce-10acre-ranch [hunk 1]
```diff
@@ -54,8 +57,11 @@ WAIT=true
 INSTANCE_COUNT=3
 OS_FAMILY=ubuntu-14-04
 DOCKER_URL="https://get.docker.com/"
+MACHINE_TYPE="n1-standard=1"
```
Comment (cloudnautique) [inline bin/gce-10acre-ranch]: n1-standard-1

Comment (cloudnautique) [inline bin/gce-10acre-ranch]: Actually would you mind setting it to f1-micro please =D

**Comment by coleca** [comment]
Updating based on feedback of previous PR

**Comment by cloudnautique** [comment]
Thanks for updating! 

The firewall rule creation/deletion looks like it won't work for us. All of our developers use this to launch environments within the same google project. If Dev A creates an env and then Dev B creates an env, and Dev A deletes their environment Dev B will be impacted. I like the idea of creating the rule in the script as a one stop solution, but I feel like adding the options to do so would be to much. 

I'm inclined to remove the fw create/delete and add one more option for the tags? Leaving the user to have it already created ahead of time. Thoughts? Alternate ideas :) I'm open.

**Comment by coleca** [comment]
I can pull the firewall piece out.  The only other thing I can think of would be to do something like Kubernetes does in their clusters/gce/util.sh script where they check for the presence of a rule first.  However they are just checking by name and we wouldn't know the names of the existing rules in advance because the devs could have created rules with any name.  

I've also discovered that it's a bad practice to have 2 duplicate firewall rules with different names.  GCE tends to drop packets here and there if you do that (experience speaking). 

All I can think of would be a little echo statement with the firewall create command in it and a message like "Make sure firewall ports 80, etc. are open for tags XX if necessary, use command -- gcloud firewall create xxx" at the end of the build_cluster function.

**Comment by cloudnautique** [comment]
OK, lets pull out the creation parts. 

Unfortunately, we need to stick with the names that exist. I like the idea of checking and if it doesn't exist echo out the statement create a open-compute-node tag with at least UDP 500 4500 ports open. 

Once we get that, I'll run the script for a day or so. (I build and teardown environments lots during the day :)) 

Thanks for working through this!

**Comment by coleca** [comment]
I removed the FW create stuff.  I played around with detecting the existence of the rule and creating the message, but it felt way too hacky/fragile and dependent on Google not changing the text output of the command.

**Comment by cloudnautique** [comment]
Cool, thanks for giving it a shot. I'll start using it.

**Comment by cloudnautique** [comment]
OK, I have been using this for a bit and it seems to be working. Sorry, would you mind changing the default to g1-small the f1-micro is useless :) 

I'll merge after that. Thanks for taking the time with this!

**Comment by coleca** [comment]
All set.. Thanks!

**Pull request merged**